iroh-rpc/src/core.rs

```diff
@@ -0,0 +1,567 @@
+                        let _ = self
+                            .pending_requests
+                            .remove(&PendingId::RequestId(request_id))
+                            .expect("Request to still be pending.")
+                            .send(SenderType::Ack);
+                    }
+                    StreamingResponseEvent::StreamError(e) => {
+                        debug!(target: "inbound streaming", "Received Ack for request {}", request_id);
+                        let _ = self
+                            .pending_requests
+                            .remove(&PendingId::RequestId(request_id))
+                            .expect("Request to still be pending.")
+                            .send(SenderType::Error(e.into()));
+                    }
+                },
+            },
+            StreamingEvent::OutboundFailure {
+                request_id, error, ..
+            } => {
+                let _ = self
+                    .pending_requests
+                    .remove(&PendingId::RequestId(request_id))
+                    .expect("Request to still be pending.")
+                    .send(SenderType::Error(Box::new(error)));
+            }
+            StreamingEvent::InboundFailure {
+                peer,
+                request_id,
+                error,
+            } => {
+                print!("\n\nThis peer_id: {}\n", self.swarm.local_peer_id());
+                println!(
+                    "RequestResponseEvent::InboundFailure:\npeer_id: {}\nrequest_id: {}\nerror: {}",
+                    peer, request_id, error
+                );
+            }
+            StreamingEvent::ResponseSent { .. } => {}
+        }
+    }
+}
+
+pub struct InboundEvent {
+    pub command: InCommand,
+    pub channel: StreamingResponseChannel,
+}
+
+// eventually generalize, pass in config etc
+pub fn new_mem_swarm(id_keys: Keypair) -> Swarm<CoreBehaviour> {
+    let peer_id = id_keys.public().to_peer_id();
+    SwarmBuilder::new(mem_transport(id_keys), CoreBehaviour::new(), peer_id).build()
+}
+
+pub async fn new_swarm(id_keys: Keypair) -> Result<Swarm<CoreBehaviour>, Box<dyn Error>> {
+    let peer_id = id_keys.public().to_peer_id();
+    Ok(SwarmBuilder::new(
+        libp2p::development_transport(id_keys).await?,
+        CoreBehaviour::new(),
+        peer_id,
+    )
+    .build())
+}
+
+use libp2p::core::transport::{MemoryTransport, Transport};
+use libp2p::core::{muxing, transport, upgrade};
+use libp2p::mplex;
+use libp2p::noise;
+use libp2p::yamux;
+
+pub fn mem_transport(keypair: Keypair) -> transport::Boxed<(PeerId, muxing::StreamMuxerBox)> {
+    let noise_keys = noise::Keypair::<noise::X25519Spec>::new()
+        .into_authentic(&keypair)
+        .unwrap();
+    MemoryTransport::default()
+        .upgrade(upgrade::Version::V1)
+        .authenticate(noise::NoiseConfig::xx(noise_keys).into_authenticated())
```

Comment:
no need for encryption on the memory transport

Reply:
if you switch to plaintext it shaves off another ~20%

Reply:
https://docs.rs/libp2p/latest/libp2p/plaintext/struct.PlainText2Config.html

Reply:
omg rock star thank you.